src/python/pants/option/options_bootstrapper.py

```diff
@@ -229,15 +232,18 @@ def _full_options(self, known_scope_infos: FrozenOrderedSet[ScopeInfo]) -> Optio
 
         return options
 
-    def full_options_for_scopes(self, known_scope_infos: Iterable[ScopeInfo]) -> Options:
+    def full_options_for_scopes(
+        self, known_scope_infos: Iterable[ScopeInfo], allow_unknown_options: bool = False
```

Comment:
Nit: consider requiring kwargs for this bool option. Imo, it makes the code easier to read in non-IDE situations.